assignments/tasks.go

```diff
@@ -0,0 +1,284 @@
+	issueOptions := &scm.CreateIssueOptions{
+		Organization: course.Name,
+		Repository:   filepath.Base(repo.GetHTMLURL()), // Needs to be of type "tests", not "https://github.com/qf101/tests". This is a very hacky solution. pb.Repository should probably have a field "Name" that is set upon creation.
+		Title:        task.Title,
+		Body:         task.Body,
+	}
+	issue, err := sc.CreateIssue(c, issueOptions)
+	if err != nil {
+		return nil, err
+	}
+	return issue, nil
+}
+
+// This is more of a converter function, and cannot currently return an error. Should probably be renamed or something.
+func createIssue(c context.Context, repo *pb.Repository, task *pb.Task, scmIssue *scm.Issue) (*pb.Issue, error) {
+	issue := &pb.Issue{
+		RepositoryID: repo.ID,
+		IssueNumber:  uint64(scmIssue.IssueNumber),
+		Name:         task.Name,
+		Title:        task.Title,
+		Body:         task.Body,
+	}
+	return issue, nil
+}
+
+// Following is Oje code (placement might be temporary):
+
+func getTasksFromAssignments(c context.Context, assignments []*pb.Assignment) []*pb.Task {
+	tasks := []*pb.Task{}
+	for _, assignment := range assignments {
+		tasks = append(tasks, assignment.Tasks...)
+	}
+
+	return tasks
+}
+
+func handleTasks(c context.Context, db database.Database, s scm.SCM, course *pb.Course, tasks []*pb.Task) error {
+	repos, err := db.GetRepositoriesWithIssues(&pb.Repository{
+		OrganizationID: course.GetOrganizationID(),
+	})
+	if err != nil {
+		return err
+	}
+
+	assignments, err := db.GetAssignmentsByCourse(course.GetID(), false)
+	if err != nil {
+		return err
+	}
+
+	// Loops through all assignments
+	for _, assignment := range assignments {
+		err := synchronizeTasks(c, db, assignment, tasks)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Loops through all student repos
+	for _, repo := range repos {
+		if !repo.IsStudentRepo() {
+			continue
+		}
+
+		err = synchronizeIssues(c, db, course, s, repo, tasks)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// synchronizeTasks synchronizes tasks in the database, with the ones of given assignment.
+// Returns a slice of tasks as they appear in the database.
+func synchronizeTasks(c context.Context, db database.Database, assignment *pb.Assignment, tasks []*pb.Task) error {
+	tasksToBeCreated := []*pb.Task{}
+	tasksToBeUpdated := []*pb.Task{}
+	taskMap := make(map[string]*pb.Task)
+	for _, task := range tasks {
+		if task.AssignmentOrder == assignment.Order {
+			taskMap[task.Name] = task
+		}
+	}
+
+	existingTasks, err := db.GetTasks(&pb.Task{AssignmentID: assignment.GetID()})
+	if err != nil {
+		return err
+	}
+
+	for _, existingTask := range existingTasks {
+		task, ok := taskMap[existingTask.Name]
+		if !ok {
+			// There exists a task in db, that is not represented by a task found in scm.
+			db.DeleteTask(existingTask)
+			continue
+		}
+		if !(task.Title == existingTask.Title && task.Body == existingTask.Body) {
```

Comment:
This boolean expression can be simplified using De Morgan's laws:
```go
	if task.Title != existingTask.Title || task.Body != existingTask.Body {
```
But perhaps you should move this check to a method on `Task` in the `ag` package:
```go
func (t *Task) HasChanged(new *Task) bool {
     return t.Title != new.Title || t.Body != new.Body
}
```
Should also add a simple test case for this `HasChanged()` method.